### PR TITLE
Add config params support for create and update command

### PR DESCRIPTION
**Pull request by sarmadsangi**
@zendesk/vegemite 

## Description

User reported an issue with our update command where it does not take provided config params in when updating an app. I did a little digging and found out that we didn't fully ported this feature over for `create` and `update` commands.

REF:
https://github.com/zendesk/zcli/issues/14
https://develop.zendesk.com/hc/en-us/articles/360050791614/comments/1260800244929

## Detail

This PR re-uses `apps:server` command `getAppSettings` for `apps:create` and `apps:update`. Along the way, i also did a little clean up.

## Checklist

- [x] :guardsman: includes new unit and functional tests

### CODE REVIEW

**Comment by sarmadsangi** [comment]
Functional tests are updated, i ll do a manual QA before release to make sure its all good.

File: packages/zcli-apps/src/commands/apps/create.ts [hunk 1]
```diff
@@ -46,7 +48,11 @@ export default class Create extends Command {
       try {
         const { app_id }: any = await getUploadJobStatus(job_id, appPath)
         cli.action.stop('Deployed')
-        const settings = manifest.parameters ? await promptAndGetSettings(manifest.parameters) : {}
+
+        const allConfigs = getAllConfigs(appPath)
+        const configParams = allConfigs?.parameters || {} // if there are no parameters in the config, just attach an empty object
+
+        const settings = manifest.parameters ? await getAppSettings(manifest, configParams) : {}
```
Comment (token-cjg) [inline packages/zcli-apps/src/commands/apps/create.ts:55]: Have you covered the case when `configParams` is `{}` here?

**Review by token-cjg — APPROVED** [review]

File: packages/zcli-apps/src/commands/apps/create.ts (hunk 1 above)
Comment (sarmadsangi) [reply packages/zcli-apps/src/commands/apps/create.ts:55]: Just checked, yes so we have two apps one with no config params and one with params. Functional tests run against both, so they are covered.

**Comment by sarmadsangi** [comment]
I have done some basic dev QA with params in config file and without params in config file for both update and create, it works as expected.
```
yarn dev apps:create ../demo_apps/v2/support/secure_settings_sample_app
```

```
yarn dev apps:update ../demo_apps/v2/support/secure_settings_sample_app
```

**Comment by ErdaKamarudin** [comment]
Changea

**Comment by ErdaKamarudin** [comment]
File changes

**Pull request merged**